scylla/src/transport/session.rs

```diff
@@ -650,15 +650,34 @@ impl Session {
                 |connection: Arc<Connection>,
                  consistency: Consistency,
                  execution_profile: &ExecutionProfileInner| {
+                    let default_consistency = execution_profile.serial_consistency;
                     let serial_consistency = query
                         .config
                         .serial_consistency
-                        .unwrap_or(execution_profile.serial_consistency);
+                        .unwrap_or(default_consistency);
                     // Needed to avoid moving query and values into async move block
                     let query_ref = &query;
                     let values_ref = &serialized_values;
                     let paging_state_ref = &paging_state;
+                    let values_size = values.serialized().unwrap().into_owned().size();
                     async move {
+                        if values_size != 0 {
+                            let prepared = connection.prepare(query_ref).await?;
+                            let serial_consistency = prepared
+                                .config
+                                .serial_consistency
+                                .unwrap_or(default_consistency);
```

Comment:
The prepared statement will have exactly the same config as the original query object, so there is no need to calculate serial consistency again.